### ISSUE TITLE
Renaming of identical but differently-named files breaks post-processing

### ISSUE THREAD

**Issue by bambury** [sabnzbd/sabnzbd#2204]
Hi, I just updated to 3.6.0 and I'm having a quick-check-related issue, so I thought I'd post about it here.

Essentially, the NZB contains identical files with different names (because of reasons). Sab renames fileS01E01.ext to fileS01E02.ext (which I'd rather not to), then also tries to rename fileS01E01.ext to fileS01E03.ext (and so on, I'd guess) and stops with an error because of course fileS01E01.ext doesn't exist anymore.

 Debug log extract is [here](https://0bin.net/paste/PzL+GrL4#2nQVv8WYp81coUcNhI6KiiiSOZcqO-D04fEoTAj0YNG).
```
2022-06-08 10:34:53,926::INFO::[postproc:750] Running verification and repair on set <HASH>ba26ad9
2022-06-08 10:34:53,928::INFO::[newsunpack:2086] Quick-check of file the.blacklist.s01e02.bdrip.x264-demand.proof.jpg failed due to non-unique hash!
2022-06-08 10:34:53,928::INFO::[newsunpack:2129] Quick-check could not find a match for file the.blacklist.s01e02.bdrip.x264-demand.proof.jpg!
2022-06-08 10:34:53,929::DEBUG::[newsunpack:2027] Quick-check will rename the.blacklist.s01e01.bdrip.x264-demand.proof.jpg to the.blacklist.s01e02.bdrip.x264-demand.proof.jpg
2022-06-08 10:34:53,930::DEBUG::[filesystem:868] Renaming "/data/Temp/The.Blacklist.S01.BDRip.x264-DEMAND/the.blacklist.s01e01.bdrip.x264-demand.proof.jpg" to "/data/Temp/The.Blacklist.S01.BDRip.x264-DEMAND/the.blacklist.s01e02.bdrip.x264-demand.proof.jpg"
2022-06-08 10:34:53,931::INFO::[newsunpack:1125] Quick-check for <HASH>ba26ad9 is OK, skipping repair
2022-06-08 10:34:53,936::DEBUG::[filesystem:904] [sabnzbd.newsunpack.par2_repair] Deleting file /data/Temp/The.Blacklist.S01.BDRip.x264-DEMAND/the.blacklist.s01e02.bdrip.x264-demand.proof.vol0+1.par2
2022-06-08 10:34:53,937::INFO::[postproc:750] Running verification and repair on set <HASH>5bd8af5
2022-06-08 10:34:53,939::DEBUG::[newsunpack:2096] Quick-check will rename the.blacklist.s01e02.bdrip.x264-demand.proof.jpg to the.blacklist.s01e03.bdrip.x264-demand.proof.jpg
2022-06-08 10:34:53,940::DEBUG::[filesystem:868] Renaming "/data/Temp/The.Blacklist.S01.BDRip.x264-DEMAND/the.blacklist.s01e01.bdrip.x264-demand.proof.jpg" to "/data/Temp/The.Blacklist.S01.BDRip.x264-DEMAND/the.blacklist.s01e03.bdrip.x264-demand.proof.jpg"
2022-06-08 10:34:53,941::INFO::[notifier:123] Sending notification: Error - Post Processing Failed for The.Blacklist.S01.BDRip.x264-DEMAND (see logfile) (type=error, job_cat=None)
2022-06-08 10:34:53,941::ERROR::[postproc:596] Post Processing Failed for The.Blacklist.S01.BDRip.x264-DEMAND (see logfile)
2022-06-08 10:34:53,942::INFO::[postproc:597] Traceback: 
Traceback (most recent call last):
  File "/app/sabnzbd/sabnzbd/postproc.py", line 388, in process_job
    par_error, re_add = parring(nzo)
  File "/app/sabnzbd/sabnzbd/postproc.py", line 751, in parring
    need_re_add, res = par2_repair(nzo, setname)
  File "/app/sabnzbd/sabnzbd/newsunpack.py", line 1123, in par2_repair
    qc_result = quick_check_set(setname, nzo)
  File "/app/sabnzbd/sabnzbd/newsunpack.py", line 1980, in quick_check_set
    new = quick_check_set_new(setname, nzo)
  File "/app/sabnzbd/sabnzbd/newsunpack.py", line 2118, in quick_check_set_new
    elif os.path.getsize(nzf.filepath) != par2info.filesize:
  File "/usr/lib/python3.9/genericpath.py", line 50, in getsize
    return os.stat(filename).st_size
FileNotFoundError: [Errno 2] No such file or directory: '/data/Temp/The.Blacklist.S01.BDRip.x264-DEMAND/the.blacklist.s01e01.bdrip.x264-demand.proof.jpg'
2022-06-08 10:34:53,951::INFO::[notifier:123] Sending notification: Download Failed - The.Blacklist.S01.BDRip.x264-DEMAND (type=failed, job_cat=*)

```

NZB file is [here](https://anonfiles.com/X7X9b7oeya/The.Blacklist.S01.BDRip.x264-DEMAND_nzb).

Please let me know if any further information or testing is necessary, I'll be happy to help.

**Comment by Safihre** [comment]
Looks indeed like a bug in the new implementation. Will fix!

**Comment by bambury** [comment]
Great, thank you! :)

**Comment by Safihre** [comment]
@bambury Could you share the NZB again or email it to me at safihre@sabnzbd.org?

**Comment by bambury** [comment]
Sure thing, I didn't expect it to be removed so quickly from the other host.

[Here](https://anonymshare.com/gD21/theblacklists01bdripx264-demand.nzb) you go.

**Comment by Safihre** [comment]
Closed by 0347907044ee2a8b060c9a53ca362c8210f6e83b
